test/geos_capi/polygon_test.rb

```diff
@@ -90,7 +90,12 @@ def test_simplify_preserve_topology
     simplified = poly.simplify_preserve_topology(1)
     interior_points = simplified.interior_rings[0].points
 
-    assert_equal 5, interior_points.length
+    # https://github.com/libgeos/geos/pull/784
+    if Gem::Version.new(RGeo::Geos.version) >= Gem::Version.new("3.12.0")
```

Comment:
Maybe we add this check as a method in test_helper.rb in case we need to do this check in the future.

```ruby
def geos_version_gteq(version)
  Gem::Version.new(RGeo::Geos.version) >= Gem::Version.new(version)
end
```